aspnetcore/signalr/hubs.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Use Hubs in ASP.NET Core SignalR
+author: rachelappel
+description: Learn how to use hubs in ASP.NET Core SignalR.
+manager: wpickett
+ms.author: rachelap
+ms.custom: mvc
+ms.date: 03/23/2018
+ms.prod: aspnet-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/hubs
+---
+
+# Use Hubs in SignalR for ASP.NET Core
+
+By [Rachel Appel](https://twitter.com/rachelappel) and [Kevin Griffin](http://twitter.com/1kevgriff)
+
+## What is a SignalR Hub
+
+The SignalR Hubs API enables you to call methods on connected clients from the server. In server code, you define methods that are called by client. In client code, you define methods that are called from the server. SignalR takes care of everything behind the scenes that makes real-time client-to-server and server-to-client communications possible.
+
+## Configure SignalR hubs
+
+Since SignalR is middleware, a call to `services.AddSignalR` is required in the `ConfigureServices` method of the `Startup` class to register the service.
+
+[!code-javascript[Configure service](hubs/sample/startup.cs?range=35)]
+
+When adding SignalR functionality to an ASP.NET MVC application, setup SignalR routes by calling `app.UseSignalR` in the `Configure` method before the other routes.
+
+[!code-javascript[Configure routes to hubs](hubs/sample/startup.cs?range=55-58)]
+
+## Create and use hubs
+
+To create a hub, inherit from the `Hub` class, and add public methods to it. Clients can call methods that are defined as `public`. Methods that use access modifiers such as `private` or `protected` can't accept calls from clients.
+
+[!code-csharp[Create and use hubs](hubs/sample/hubs/chathub.cs?range=10-14)]
+
+You can specify a return type and parameters, including complex types and arrays, as you would in any C# method. Data that you receive in parameters or return to the caller is communicated between the server and the client by using JSON. SignalR handles the binding of complex objects and arrays of objects automatically.
+
+## Connection IDs
+
+Each connection to a SignalR hub has a unique connection ID. This ID can be used by other parts of your application to send messages directly to a particular connection.
+
+In the context of your hub, you can determine current by referencing its `Context.ConnectionId` object.
+
+[!code-csharp[Connection Ids](hubs/sample/hubs/chathub.cs?range=20-24)]
+
+## The Clients object
+
+Each instance of the `Hub` class has a property named `Clients` that contains the following members for communication between server and client:
+
+| Property | Description |
+| ------ | ----------- |
+| All | Calls a method on all connected clients |
+| Caller | Calls a method on all connected clients |
+| Others | Calls a method on clients other than a specific connection |
+
+Additionally, the `Hub` class contains the following methods:
+
+| Method | Description |
+| ------ | ----------- |
+| AllExcept | Calls a method on all connected clients except a specified connection |
```

Comment:
It actually accepts multiple connections, so "Calls a method on all connected clients **except** for the provided connections"